### PR TITLE
[BUG Fix]: Old issue close

### LINKED ISSUE #138: [BUG] Old issue close
**Describe the bug**
The workflow only closes issues after 30 days timeline. 

**Expected behavior**
It should close issue after PR is merged

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by yashksaini-coder**
- Fixes #138 

## 📋 Description

This GitHub Action automatically closes issues that have been inactive for 30 days. It runs daily and excludes issues with specific labels like `pinned` or `do-not-close`, ensuring important issues remain open. It also adds a comment explaining why the issue was closed, providing transparency for contributors.

## 🔨 Changes Made

- Error Handling: The try...catch block captures any errors during issue fetching or closing, providing better feedback in case of failures.
- Label Exclusions: Issues labeled with pinned or do-not-close will not be closed. You can customize this list.
- Exclusion of Pull Requests: Ensures that pull requests are not inadvertently closed.
- Activity Check: Filters issues based on their last activity date.
- Logging: Improved logging to track progress, including how many issues were found and closed.
- Comments on Closed Issues: After closing an issue, a comment is added to inform contributors about why it was closed.

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [X] My code follows the style guidelines of this project.
- [X] Read Contributing.md

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [X] Bug fix (non-breaking change which fixes an issue) 🐛
- [ ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to Talk-Trove! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@Sibam-Paul review this PR

**Comment by MastanSayyad** [comment]
@kartikayasijaa please review this PR

File: .github/workflows/close-old-pr.yml [hunk 1]
```diff
@@ -1,36 +1,70 @@
...
+              const { data: issues } = await github.issues.listForRepo({
+                owner,
+                repo,
+                state: 'open',
+                per_page: 100,  // Max per page
+                since: daysAgo.toISOString(),
               });
+
+              // Filter issues that are older than 30 days
+              const oldIssues = issues.filter(issue =>
```
Comment (kartikayasijaa) [inline .github/workflows/close-old-pr.yml:36]: add this logic when we first retrieve all the issues. add this in filter

**Comment by kartikayasijaa** [comment]
@MastanSayyad Added comments.

File: .github/workflows/close-old-pr.yml (hunk 1 above)
Comment (yashksaini-coder) [reply .github/workflows/close-old-pr.yml:36]: Hi @Sibam-Paul @kartikayasijaa ,  Thanks for being patient.
I have added the retrieve logic when retrieving all the issues as per you request

image

**Comment by MastanSayyad** [comment]
@kartikayasijaa please add me as a collaborator, we are reviewing contributors and need access to reduce there levels, thank you!

Program Manager
GSSOC Extd

File: .github/workflows/close-old-issue.yml [hunk 2]
```diff
@@ -22,10 +23,12 @@ jobs:
               owner: context.repo.owner,
               repo: context.repo.repo,
               state: 'open',
-              since: daysAgo.toISOString(),
             });
 
-            for (const issue of issues) {
+            // Filter issues that were last updated before the specified number of days
+            const oldIssues = issues.filter(issue => new Date(issue.updated_at) < daysAgo);
+
```
Comment (Sibam-Paul) [inline .github/workflows/close-old-issue.yml:30]: ok

File: .github/workflows/close-old-pr.yml (hunk 1 above)
Comment (Sibam-Paul) [reply .github/workflows/close-old-pr.yml:36]: ok

**Review by Sibam-Paul — APPROVED** [review]

**Review by yashksaini-coder — COMMENTED** [review]
New changes bcz updated profile picture

**Pull request closed without merging**